woocommerce/woocommerce
# Block Hooks API: Abstract Block Hooks code into a separate trait class

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR abstracts the logic used to auto-insert the Mini Cart into the header of select themes and moves it to the AbstractBlock class so other blocks can benefit from it.

Closes https://github.com/woocommerce/woocommerce/issues/43266
Closes https://github.com/woocommerce/woocommerce/issues/42756



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure before testing you run `pnpm install`
2. Follow the testing instructions on this PR https://github.com/woocommerce/woocommerce-blocks/pull/11745 and ensure there are no regressions introduced by this refactor.
3. Please note when testing this, it will also work for other default themes (TT2/3/4) as well as select Automattic themes.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Abstract the Block Hooks logic from the Mini Cart class to its own trait class so other blocks can benefit.

#### Comment 

</details>
